### ISSUE TITLE
Feature Request: `AWS::StepFunctions::StateMachine` - Correctly recognize the max character limit of a state name

### ISSUE THREAD

**Issue by r-heimann** [aws-cloudformation/cfn-lint#4079]
### Is this feature request related to a new rule or cfn-lint capabilities?

New capability

### Describe the feature you'd like to request

Currently cfn-lint is unable to recognize the correct character limit of a state name. The following CloudFormation Template will cause a CloudFormation error:

```yaml
AWSTemplateFormatVersion: 2010-09-09
Description: Test

Resources:
  StepFunctionRole:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Statement:
          - Effect: Allow
            Principal:
              Service: states.amazonaws.com
            Action: sts:AssumeRole

  StateMachine:
    Type: AWS::StepFunctions::StateMachine
    Properties:
      RoleArn: !GetAtt StepFunctionRole.Arn
      Definition:
        StartAt: Passsssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss
        States:
          Passsssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss:
            Type: Pass
            Next: Success
          Success:
            Type: Succeed
```

CloudFormation error:

```
Resource handler returned message:
"Invalid State Machine Definition: 'INVALID_STATE_NAME: Invalid State name:
State 'Passsssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss' exceeds the 80-character limit allowed by the service at
/States/Passsssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss'
(Service: Sfn, Status Code: 400, Request ID: 6...8)
(SDK Attempt Count: 1)"
(RequestToken: 9...4, HandlerErrorCode: InvalidRequest)
```

### Describe the solution you'd like

It would be great if cfn-lint could give an error on this.

### Additional context

cfn-lint seems to have a different check though:

```yaml
AWSTemplateFormatVersion: 2010-09-09
Description: Test

Resources:
  StepFunctionRole:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Statement:
          - Effect: Allow
            Principal:
              Service: states.amazonaws.com
            Action: sts:AssumeRole

  StateMachine:
    Type: AWS::StepFunctions::StateMachine
    Properties:
      RoleArn: !GetAtt StepFunctionRole.Arn
      Definition:
        StartAt: Passsssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss
        States:
          Passsssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss:
            Type: Pass
            Next: Success
          Success:
            Type: Succeed
```

causes

```
[cfn-lint] E3601: 'Passsssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss' does not match any of the regexes: '^.{1,128}$'
```

### Is this something that you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request

### Would this feature include a breaking change?

- [ ] ⚠️ This feature might incur a breaking change
